test/gtest/prelu.hpp

```diff
@@ -0,0 +1,213 @@
+    tcs.push_back({{512, 64, 112, 112}, false});
+    tcs.push_back({{512, 64, 56, 56}, false});
+    tcs.push_back({{512, 128, 56, 56}, false});
+    tcs.push_back({{512, 128, 28, 28}, false});
+    tcs.push_back({{512, 256, 28, 28}, false});
+
+    tcs.push_back({{512, 64, 112, 112}, true});
+    tcs.push_back({{512, 64, 56, 56}, true});
+    tcs.push_back({{512, 128, 56, 56}, true});
+    tcs.push_back({{512, 128, 28, 28}, true});
+    tcs.push_back({{512, 256, 28, 28}, true});
+
+    return tcs;
+}
+
+inline std::vector<size_t> GetStrides(std::vector<size_t> lengths, bool contiguous)
+{
+    if(!contiguous)
+        std::swap(lengths.front(), lengths.back());
+    std::vector<size_t> strides(lengths.size());
+    strides.back() = 1;
+    for(int i = lengths.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * lengths[i + 1];
+    if(!contiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename T = float>
+struct PReLUTest : public ::testing::TestWithParam<PReLUTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle   = get_handle();
+        prelu_config    = GetParam();
+        auto gen_value1 = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-6, 100); };
+        auto gen_value2 = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-6, 99); };
+
+        auto lengths = prelu_config.lengths;
+
+        auto input_strides = GetStrides(lengths, true);
+        input              = tensor<T>{lengths, input_strides}.generate(gen_value1);
+        dinput             = tensor<T>{lengths, input_strides};
+        ref_dinput         = tensor<T>{lengths, input_strides};
+        std::fill(dinput.begin(), dinput.end(), static_cast<T>(0.0f));
+        std::fill(ref_dinput.begin(), ref_dinput.end(), static_cast<T>(0.0f));
+
+        std::vector<size_t> weight_length = {1};
+        if(prelu_config.full_params)
+            weight_length = {lengths[1]};
+        auto weight_strides = GetStrides(weight_length, true);
+        weight              = tensor<T>{weight_length, weight_strides}.generate(gen_value2);
+        dweight             = tensor<T>{weight_length, weight_strides};
+        ref_dweight         = tensor<T>{weight_length, weight_strides};
+        std::fill(dweight.begin(), dweight.end(), static_cast<T>(0.0f));
+        std::fill(ref_dweight.begin(), ref_dweight.end(), static_cast<T>(0.0f));
+
+        auto out_strides = GetStrides(lengths, true);
+        doutput          = tensor<T>{lengths, out_strides};
+        std::fill(doutput.begin(), doutput.end(), static_cast<T>(1.0f));
+
+        ws_sizeInBytes = miopen::GetPReLUBackwardWorkspaceSize(handle, input.desc, weight.desc);
+        if(ws_sizeInBytes == static_cast<size_t>(-1))
+            GTEST_SKIP();
+
+        if(ws_sizeInBytes != 0)
+        {
+            std::vector<size_t> workspace_dims;
+            workspace_dims.push_back(ws_sizeInBytes / sizeof(T));
```

Comment:
T->float
The workspace is always assumed to be a float.

Reply:
You are right. This is a protential bug. I have fixed it. Thank you.